bentoml/metrics.py

```diff
@@ -0,0 +1,36 @@
+from simple_di import inject
+from simple_di import Provide
+from typing import TYPE_CHECKING
+import logging
+from ._internal.configuration.containers import BentoMLContainer
+
+logger = logging.getLogger(__name__)
+
+if TYPE_CHECKING:
+  from ._internal.server.metrics.prometheus import PrometheusClient
+
+def log_warnings(kwargs):
+  logger.warning ("registry cannot be in passed as argument")
+  if "registry" in kwargs:
+    kwargs.pop("registry")
```

Comment:
Let's not do this. Log function shouldn't mutate a state of an object.

I think for this case it is fine to duplicate the warning in each function.

Reply:
Done ✅